code/modules/mod/mod_ui.dm

```diff
@@ -84,8 +84,13 @@
 		return
 	switch(action)
 		if("lock")
-			locked = !locked
-			balloon_alert(usr, "[locked ? "locked" : "unlocked"]!")
+			if(locked & allowed(usr) || !locked)
+				req_access = wearer.get_access()
```

Comment:
This isn't doing anything

Reply:
88 sets the req_access which is what's checked when attempting to activate the modsuit.

Reply:
We're not supposed to be updating the access here though, here is only flipping whether it's locked.

Reply:
just dropping by to say that you are using the wrong comparison. That's the bitwise comparison not the logical comparison

Reply:
You're making it update the access upon locking/unlocking, that should not be the case. Access being set is in a separate place, the only thing we're supposed to do is change the status on whether it is locked, NOT set ``req_access`` to anything.

Reply:
Ok i yield access will not be set when the modsuit user locks the modsuit.